src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/EnumerableExtensions.cs

```diff
@@ -496,4 +550,382 @@ static T[] BuildSlow(IEnumerable<T> sequence)
             return builder.ToArray();
         }
     }
+
+    /// <summary>
+    ///  Projects each element of an <see cref="IEnumerable{T}"/> into a new form and sorts them in ascending order.
+    /// </summary>
+    /// <typeparam name="T">The type of the elements in <paramref name="sequence"/>.</typeparam>
+    /// <typeparam name="TResult">The type of the value returned by <paramref name="selector"/>.</typeparam>
+    /// <param name="sequence">An <see cref="IEnumerable{T}"/> of elements to invoke a transform function on and sort.</param>
+    /// <param name="selector">A transform function to apply to each element.</param>
+    /// <returns>
+    ///  Returns a new <see cref="ImmutableArray{T}"/> whose elements are the result of invoking the transform function
+    ///  on each element of <paramref name="sequence"/> and sorted in ascending order.
+    /// </returns>
+    public static ImmutableArray<TResult> SelectAndOrderAsArray<T, TResult>(this IEnumerable<T> sequence, Func<T, TResult> selector)
+    {
+        if (sequence is ImmutableArray<T> array)
+        {
+            return ImmutableArrayExtensions.SelectAndOrderAsArray(array, selector);
+        }
+
+        if (sequence is IReadOnlyList<T> list)
+        {
+            return ReadOnlyListExtensions.SelectAndOrderAsArray(list, selector);
+        }
```

Comment:
If `SelectAsArray` is already doing this switching, is it actually beneficial to repeat it here?

Reply:
That's a fair point, and I likely went a little overboard with this. :smile: